src/wallet/wallet2.cpp

```diff
@@ -6310,6 +6705,21 @@ void wallet2::store_to(const std::string &path, const epee::wipeable_string &pas
       }
     }
   }
+  else if (m_background_sync_type == BackgroundSyncCustomPassword && m_background_syncing && !m_is_background_wallet)
+  {
+    // We're background syncing, so store the wallet cache as a background cache
+    // keeping the background sync data
+    try
+    {
+      THROW_WALLET_EXCEPTION_IF(!m_custom_background_key, error::wallet_internal_error, "Custom background key not set");
+      store_background_cache(m_custom_background_key.get(), false/*do_reset_background_sync_data*/);
```

Comment:
Sensitive user data from the cache + the `m_cache_key` have been wiped by this point (grep `m_cache_key.scrub()`). For all intents and purposes, it's as if the opened wallet *is* the background wallet at this point. As long as the main wallet processes the background cache on reopen, which it will attempt to do automatically via `process_background_cache_on_open`, no progress gets lost.

If you look at the test in `transfer.py` with the comment `# Close wallet while background syncing, then reopen`, the custom password test is testing this case to make sure no progress is lost.